builder/digest_docker.go

```diff
@@ -0,0 +1,92 @@
+package builder
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"log"
+	"strings"
+
+	"github.com/Azure/acr-builder/pkg/image"
+	"github.com/Azure/acr-builder/pkg/procmanager"
+	"github.com/Azure/acr-builder/util"
+	"github.com/pkg/errors"
+)
+
+type dockerStoreDigest struct {
+	procManager *procmanager.ProcManager
+	debug       bool
+}
+
+func NewDockerStoreDigest(procManager *procmanager.ProcManager, debug bool) *dockerStoreDigest {
+	return &dockerStoreDigest{
+		procManager: procManager,
+		debug:       debug,
+	}
+}
+
+var _ DigestHelper = &dockerStoreDigest{}
+
+func (d *dockerStoreDigest) PopulateDigest(ctx context.Context, reference *image.Reference) error {
+	if reference == nil {
+		return nil
+	}
+	if reference.Digest != "" {
+		return nil
+	}
+	// refString will always have the tag specified at this point.
+	// For "scratch", we have to compare it against "scratch:latest" even though
+	// scratch:latest isn't valid in a FROM clause.
+	if reference.Reference == NoBaseImageSpecifierLatest {
```

Comment:
How does `digest_remote` handle the `scratch` base image?

Reply:
Added that